lib/puppet/parser/script_compiler.rb

```diff
@@ -36,6 +36,7 @@ def with_context_overrides(description = '', &block)
   #
   def compile
     Puppet[:strict_variables] = true
+    Puppet[:strict] = :error
```

Comment:
Is it enough to set this here? I would assume that strict is needed during parsing and validation too. And perhaps other initialization. Should the setting be controlled by a callback in the setting of `Puppet[:tasks]`?

Reply:
I tried making that change, but could not get it to work since the reverse `tasks=false` does not have a defined mapping back to the setting of strict and strict_variables because when Puppet[:tasks] = false, there is no way to know what to set them to (the values they had before tasks was set to true?, or the defaults? Gets really complicated.